File: cameraview/src/main/java/com/base/cameraview/frame/ImageFrameManager.java
package com.base.cameraview.frame;

import android.media.Image;

import androidx.annotation.NonNull;

public class ImageFrameManager extends FrameManager<Image> {

    public ImageFrameManager(int poolSize) {
        super(poolSize, Image.class);
    }

    @Override
    protected void onFrameDataReleased(@NonNull Image data, boolean recycled) {
        try {
            data.close();
        } catch (Exception ignore) {
        }
    }

    @NonNull
    @Override
    protected Image onCloneFrameData(@NonNull Image data) {
        throw new RuntimeException("Cannot freeze() an Image Frame. " +
                "Please consider using the frame synchronously in your process() method, " +
                "which also gives better performance.");
    }
}

File: cameraview/src/main/java/com/base/cameraview/frame/ImageFrameManager.kt
package com.base.cameraview.frame

import android.media.Image

class ImageFrameManager(poolSize: Int) : FrameManager<Image?>(
    poolSize,
    Image::class.java as Class<Image?>
) {
    override fun onFrameDataReleased(data: Image?, recycled: Boolean) {
        try {
            data?.close()
        } catch (ignore: Exception) {
        }
    }

    override fun onCloneFrameData(data: Image?): Image {
        throw RuntimeException(
            "Cannot freeze() an Image Frame. " +
                    "Please consider using the frame synchronously in your process() method, " +
                    "which also gives better performance."
        )
    }
}
